sql/core/src/main/scala/org/apache/spark/sql/execution/subquery.scala

```diff
@@ -194,6 +257,19 @@ case class PlanSubqueries(sparkSession: SparkSession) extends Rule[SparkPlan] {
         }
         val executedPlan = new QueryExecution(sparkSession, query).executedPlan
         InSubqueryExec(expr, SubqueryExec(s"subquery#${exprId.id}", executedPlan), exprId)
+      case expressions.Exists(sub, children, exprId) =>
```

Comment:
We should simply throw exception for any other `SubqueryExpression`, explicitly saying that it's not supported.

We can think more about how to solve this problem in your original PR.

Reply:
> We can think more about how to solve this problem in your original PR.

See current change, don't collect data, only judge has result, may don't have oom problem

Reply:
> We can think more about how to solve this problem in your original PR.

Try some ways, result is not the same as PostgresSQL.
Since for full outer join, we can't push down or build a new join.